### PR TITLE
IGAPP-496: Connect snackbar to redux state

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
image

**Review by Taggotty — APPROVED** [review]

**Comment by Taggotty** [comment]
Snackbar 🥳  
Tested on Android Emulator.

**Review by SteffiStuffel — APPROVED** [review]
Code looks good. 

How can i test the PermissingSnackbarContainer or the SnackbarContainer? When do they appear?

**Comment by klinzo** [comment]
> Code looks good.
> 
> How can i test the PermissingSnackbarContainer or the SnackbarContainer? When do they appear?

Permission snackbars are shown on the landing page if nearby locations are enabled but the permission wasn't granted. Same for iOS for push notifications on the dashboard.

File: native/src/modules/app/StoreActionType.js [hunk 1]
```diff
@@ -341,7 +341,20 @@ export type SetResourceCacheUrlActionType = {|
   +params: {| +url: string |}
 |}
 
+// Pushes a new snackbar to the queue
+export type PushSnackbarActionType = {|
+  type: 'PUSH_SNACKBAR',
```
Comment (maxammann) [inline native/src/modules/app/StoreActionType.js:346]: Should we maybe use `SET_SNACKBAR` here? There can only be one snackbar visible, according to the material design guidelines I suppose. Also `PUSH` is used with our action terminology already in a very specific sense.

Comment (maxammann) [reply native/src/modules/app/StoreActionType.js:346]: If we want to keep the array of snackbars, then I would suggest `APPEND` which is a synonym

**Review by maxammann — APPROVED** [review]

**Comment by SteffiStuffel** [comment]
> > Code looks good.
> > How can i test the PermissingSnackbarContainer or the SnackbarContainer? When do they appear?
> 
> Permission snackbars are shown on the landing page if nearby locations are enabled but the permission wasn't granted. Same for iOS for push notifications on the dashboard. The first new snackbar (via SnackbarContainer) is added with #239.

Oke thanks. Tested it, works as expected.

File: native/src/modules/app/StoreActionType.js (hunk 1 above)
Comment (klinzo) [reply native/src/modules/app/StoreActionType.js:346]: I'd like to keep the array, but `APPEND` would work for me. Would we stick with `POP` or e.g. use `REMOVE` @maxammann ?

Comment (klinzo) [reply native/src/modules/app/StoreActionType.js]: Renamed to enqueue/dequeue as push/pop/FILO is really used in terms of stacks while this is a queue/FIFO.

**Review by Taggotty — APPROVED** [review]

**Pull request merged**